### PR TITLE
feat(TFD-4068): Add multiples processing component

### PR DESCRIPTION
**Pull request by pbailly**
This PR contains the code for:
TFD-4068: Field Selector Component.
TFD-4603: Filter Component.
TFD-4610: Python Component.

### CODE REVIEW

File: processing-beam/src/main/java/org/talend/components/processing/ProcessingErrorCode.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2015 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/ProcessingErrorCode.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/ConditionsRowConstant.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/ConditionsRowConstant.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/Filter.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/Filter.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/Filter.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            @Output(REJECT_CONNECTOR) final OutputEmitter<JsonObject> reject) {
+        // no-op
+    }
+
+    @Override
+    public PCollectionTuple expand(PCollection<IndexedRecord> input) {
+        return input.apply(ParDo.of(new FilterDoFn(configuration)).withOutputTags(flowOutput, TupleTagList.of(rejectOutput)));
+    }
+
+    // TODO: Currenlty the component alway send data on output and reject link, even if they are not
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/Filter.java]:
```suggestion
    // TODO: Currently the component always sends data on output and reject links, even if they are not
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/FilterDoFn.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/FilterDoFn.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/FilterPredicate.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/FilterPredicate.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/FilterUtils.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/FilterUtils.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/LogicalOpType.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/LogicalOpType.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/main/java/org/talend/components/processing/filter/TypeConverterUtils.java [hunk 9]
```diff
@@ -0,0 +1,231 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/main/java/org/talend/components/processing/filter/TypeConverterUtils.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/test/java/org/talend/components/processing/filter/FilterDoFnAvpathTest.java [hunk 10]
```diff
@@ -0,0 +1,379 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/test/java/org/talend/components/processing/filter/FilterDoFnAvpathTest.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/test/java/org/talend/components/processing/filter/FilterDoFnTest.java [hunk 11]
```diff
@@ -0,0 +1,697 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/test/java/org/talend/components/processing/filter/FilterDoFnTest.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/test/java/org/talend/components/processing/filter/TypeConverterUtilsTest.java [hunk 12]
```diff
@@ -0,0 +1,459 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2017 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/test/java/org/talend/components/processing/filter/TypeConverterUtilsTest.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/src/test/java/org/talend/components/processing/python/PythonDoFnTest.java [hunk 13]
```diff
@@ -0,0 +1,254 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2016 Talend Inc. - www.talend.com
```
Comment (rdubois) [inline processing-beam/src/test/java/org/talend/components/processing/python/PythonDoFnTest.java]:
```suggestion
// Copyright (C) 2006-2018 Talend Inc. - www.talend.com
```

File: processing-beam/pom.xml [hunk 14]
```diff
@@ -62,14 +82,43 @@
...
     <!-- guava for components-adapter-beam -->
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
       <version>23.0</version>
     </dependency>
+
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-all</artifactId>
```
Comment (rdubois) [inline processing-beam/pom.xml]: There is already a `harmcrest-junit` dependency in the the root pom.xml, I guess this one can be removed. WDYT ?

File: connectors-se-docker/pom.xml [hunk 15]
```diff
@@ -47,6 +47,24 @@
       <artifactId>fileio</artifactId>
       <version>${project.version}</version>
       <scope>runtime</scope>
+      <exclusions>
```
Comment (rdubois) [inline connectors-se-docker/pom.xml]: Still wondering why we need these exclusions. Can you please leave few comments in the PR to explain why ?

File: processing-beam/pom.xml [hunk 16]
```diff
@@ -62,14 +82,43 @@
           <groupId>org.apache.beam</groupId>
           <artifactId>beam-sdks-java-core</artifactId>
         </exclusion>
+        <exclusion>
```
Comment (rdubois) [inline processing-beam/pom.xml]: Still wondering why we need these exclusions. Can you please leave few comments in the PR to explain why ?

File: connectors-se-docker/pom.xml (hunk 15 above)
Comment (pbailly) [reply connectors-se-docker/pom.xml]: We currently import TCOMP v0, that import beam 2.5.
Having the dependencies will not break anything, but we still going to load too many dependencies for nothing.
Putting these exclusions will make the different builds lighter.

File: processing-beam/pom.xml (hunk 14 above)
Comment (pbailly) [reply processing-beam/pom.xml]: No, the harmcrest-junit is not enough.

**Pull request merged**